libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/RouteRefreshController.kt

```diff
@@ -1,79 +1,141 @@
 package com.mapbox.navigation.core.routerefresh
 
+import com.mapbox.api.directions.v5.models.RouteLeg
+import com.mapbox.navigation.base.internal.route.refreshRoute
+import com.mapbox.navigation.base.internal.time.parseISO8601DateToLocalTimeOrNull
 import com.mapbox.navigation.base.route.NavigationRoute
 import com.mapbox.navigation.base.route.NavigationRouterRefreshCallback
 import com.mapbox.navigation.base.route.NavigationRouterRefreshError
 import com.mapbox.navigation.base.route.RouteRefreshOptions
 import com.mapbox.navigation.core.directions.session.DirectionsSession
-import com.mapbox.navigation.core.trip.session.TripSession
-import com.mapbox.navigation.utils.internal.MapboxTimer
-import com.mapbox.navigation.utils.internal.ThreadController
 import com.mapbox.navigation.utils.internal.logE
 import com.mapbox.navigation.utils.internal.logI
 import com.mapbox.navigation.utils.internal.logW
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.async
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlinx.coroutines.withTimeoutOrNull
+import java.util.Date
+import kotlin.coroutines.resume
 
 /**
  * This class is responsible for refreshing the current direction route's traffic.
  * This does not support alternative routes.
- *
- * If the route is successfully refreshed, this class will update the [DirectionsSession].
  */
 internal class RouteRefreshController(
     private val routeRefreshOptions: RouteRefreshOptions,
     private val directionsSession: DirectionsSession,
-    private val tripSession: TripSession,
-    threadController: ThreadController,
+    private val currentLegIndexProvider: () -> Int,
     private val routeDiffProvider: DirectionsRouteDiffProvider = DirectionsRouteDiffProvider(),
+    private val localDateProvider: () -> Date
 ) {
 
     internal companion object {
         internal const val LOG_CATEGORY = "RouteRefreshController"
     }
 
-    private val routerRefreshTimer = MapboxTimer(threadController).apply {
-        restartAfterMillis = routeRefreshOptions.intervalMillis
+    suspend fun refresh(routes: List<NavigationRoute>): List<NavigationRoute> {
+        return if (routes.firstOrNull()?.routeOptions?.enableRefresh() == true) {
```

Comment:
just curious why you check the same conditions here and `L117-118`

Reply:
thanks, got rid of redundant checks